File: solutions/ruby/game-of-thrones-1.rb
def has_palindrome(string)
  letters = string.chars
  temp_array = [];
  letters.each { |letter| temp_array.include?(letter) ? temp_array.delete(letter) : temp_array.push(letter) }

  temp_array.size > 1 ? "NO" : "YES"
end

p has_palindrome("aaabbbb");
p has_palindrome("cdefghmnopqrstuvw");
p has_palindrome("cdcdcdcdeeeef");

File: solutions/javascript/game-of-thrones-1.js
function has_palindrome(string) {
  var letters = string.split('');
  var tempArray = [];

  letters.forEach(function(letter){
    if (tempArray.indexOf(letter) > -1) {
      tempArray.splice(tempArray.indexOf(letter), 1);
    } else {
      tempArray.push(letter);
    }
  });

  return tempArray.length > 1 ? "NO" : "YES";
}

console.log(has_palindrome("aaabbbb"));
console.log(has_palindrome("cdefghmnopqrstuvw"));
console.log(has_palindrome("cdcdcdcdeeeef"));
